### ISSUE TITLE
Underestimation of alignment end position for long-reads

### ISSUE THREAD

**Issue by Adamtaranto** [tseemann/samclip#5]
Samclip calculates alignment end position as alignment start position + length of read. 

`my $end = $start + length($sam[SAM_SEQ]) - 1;`

This works fine for Illumina data, but often falls short of the true alignment length when dealing with long-reads that may contain many deletions relative to the reference. I expect that this will cause samclip to falsely exclude some long read alignments which are actually soft clipped at the 3' end of contigs.

I fixed this in teloclip by calculating alignment len (in reference) directly from the CIGAR string.

**Comment by tseemann** [comment]
This tool was only designed for short reads really - and I hadn't considerd your use case. But you are exactly right!  I will have a look at `teloclip` - I assume you just correct for `I` and `D` tags.

**Comment by Adamtaranto** [comment]
Yep, also potential splices and mismatches. See [lenCIGAR](https://github.com/Adamtaranto/teloclip/blob/master/teloclip/__init__.py) function.

**Comment by tseemann** [comment]
Ah yes, the infamous `X` and `=` operators. I've never seen them used in practice. Do any of the nanopore tools use them?  For short reads they would make the SAM files way too big.

**Comment by Adamtaranto** [comment]
I haven't seen them in any of my data but figured I should support them just in case.
